src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -168,14 +173,19 @@ function _is_action_fit(method, statement) {
     return statement.Action ? action_fit : !action_fit;
 }
 
-function _is_principal_fit(account, statement) {
+function _is_principal_fit(account, statement, ignore_anon_principal = false) {
```

Comment:
why not call it "disallow_public_access" as in the parts of the flow? I think this name is more clear; we know why we should ignore the fit.

Reply:
I'm just not sure I fully get the flow. If we want to disallow public access. Shouldn't we have it only for `Allow` statement and not for `Deny`? It seems to me that we should not ignore a fit in case of a Deny statement.
Also, If the issue is with semantics, we need a better name for this as * is not just for anonymous principals it is for any principal. 
Maybe some more comments in this area will be good to make me and future code readers better understand its use.

Reply:
@tangledbytes is this relevant also for deny?

Reply:
Ah! I am actually sorry. I actually thought that I responded here but turns out I forgot. Yes, you are right, I should not ignore it in case of `DENY`.

Also yes, it sounds right that `anon` isn't the right word. I will replace it with `public_principla`? How's that?

Again. apologies, missed this 😅